src/media-groups.js

```diff
@@ -658,17 +663,28 @@ export const activeGroup = (type, settings) => (track) => {
 
   let variants = null;
 
+  // set to variants to main media active group
   if (media.attributes[type]) {
     variants = groups[media.attributes[type]];
   }
 
-  variants = variants || groups.main;
+  const groupKeys = Object.keys(groups);
+
+  if (!variants && groupKeys.length === 1) {
+    // use the main group if it exists
+    if (groups.main) {
+      variants = groups.main;
+    // only one group, use that one
+    } else if (groupKeys.length === 1) {
+      variants = groups[groupKeys[0]];
+    }
+  }
```

Comment:
It looks like there's a conditional within a repeated check within the conditional. Should the `groupKeys.length === 1` be removed from the outer conditional?